### ISSUE TITLE
Date with year 1988 doesn't work correctly when setting year

### ISSUE THREAD

**Issue by daxhuiberts** [moment/moment#4274]
**Description of the Issue and Steps to Reproduce:**

When I have a moment object of which the year is 1988 (e.g. `moment("1988-02-10")`) and I change the year (e.g. `date.year(2017)`). Then the day of month is always changed to the last day of the month.

**Environment:**

Chrome Version 61.0.3163.100 (Official Build) (64-bit) on MacOS Sierra Version 10.12.6.
Examples can be run on the javascript console on the momentjs homepage.

**Some incorrect examples::**

```js
moment("1988-02-10").year(1999).toString() // => "Sun Feb 28 1999 00:00:00 GMT+0100"
moment("1988-09-25").year(2017).toString() // => "Sat Sep 30 2017 00:00:00 GMT+0200"
```

**Some good working examples:**

```js
moment("1989-02-10").year(1999).toString() // => "Wed Feb 10 1999 00:00:00 GMT+0100"
moment("1987-09-25").year(2017).toString() // => "Mon Sep 25 2017 00:00:00 GMT+0200"
```

**Comment by emilebosch** [comment]
It happens with all leap years, not just 1988.

**Comment by emilebosch** [comment]
so the original use case was that we wanted to calculate the next birthday of the user. It yielded incorrect results

**Comment by mj1856** [comment]
Dup #4258.  Will track there.  Thanks.

**Comment by mj1856** [comment]
Actually, dup #4024 and fix pending in #4255.
